fdbclient/NativeAPI.actor.cpp

```diff
@@ -731,6 +734,24 @@ void DatabaseContext::setOption( FDBDatabaseOptions::Option option, Optional<Str
 			server_interf.clear();
 			locationCache.insert( allKeys, Reference<LocationInfo>() );
 			break;
+		case FDBDatabaseOptions::TRANSACTION_TIMEOUT:
+			transactionTimeout = extractIntOption(value, 0, std::numeric_limits<int>::max())/1000.0;
+			break;
+		case FDBDatabaseOptions::TRANSACTION_RETRY_LIMIT:
+			transactionMaxRetries = (int)extractIntOption(value, -1, std::numeric_limits<int>::max());
+			break;
+		case FDBDatabaseOptions::TRANSACTION_MAX_RETRY_DELAY:
+			validateOptionValue(value, true);
+			transactionMaxBackoff = extractIntOption(value, 0, std::numeric_limits<int32_t>::max()) / 1000.0;
+			break;
+		case FDBDatabaseOptions::SNAPSHOT_RYW_ENABLE:
+			validateOptionValue(value, false);
+			snapshotRywEnabled++;
+			break;
+		case FDBDatabaseOptions::SNAPSHOT_RYW_DISABLE:
```

Comment:
Are you aware of users of this option that would benefit from making it defaultable? Or, if not, what was the motivation for choosing this option as one that could have a database default over some of the others? I don't have a problem with it being here, I was more just curious.

Reply:
The use case I had in mind for the disable option would be for someone who wrote code against API version < 300 and then wanted to upgrade their API version (say, in order to pick up the newer versions of the Java bindings because they really want #1207) but want to preserve snapshot RYW behavior because their old code relied on it. Then the enable option is for the opposite case--someone doesn't want to upgrade their API version to 300 but wants snapshot RYW everywhere.

Reply:
Yeah, that seems fair. I don't know whether there are any people who have code that they upgraded from pre-300 and rely on this behavior, but it doesn't hurt to have this in case there are.